modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/CommunicationTcpUtils.java

```diff
@@ -201,4 +205,161 @@ public static void failNode(ClusterNode nodeToFail,
             ", err=" + err +
             ", connectErrs=" + X.getSuppressedList(err) + ']');
     }
+
+    /**
+     * Base class for network interface matchers.
+     */
+    public abstract static class NetworkInterfaceMatcher {
+        /** Pattern of network interface. */
```

Comment:
Does it make sense to document the pattern syntax somewhere?

Reply:
Updated as follows:
```
    /**
     * Base class for network interface matchers.
     *
     * An implementation of the interface should support using ranges and wildcards in the list of interfaces.
     * Wildcard symbol {@code *} represents a range of values between {@code 0} and {@code 255}. For example, {@code 12.12.12.*} refers to
     * addresses from {@code 12.12.12.0} to {@code 12.12.12.255}. Range symbol {@code -} represents a range of values. For example,
     * {@code 12.12.12.12-24} refers to addresses from {@code 12.12.12.12} to {@code 12.12.12.24}.
     */
    public abstract static class NetworkInterfaceMatcher {
```